doc/sphinx/language/core/records.rst

```diff
@@ -3,208 +3,313 @@
+
+.. table:: Printing Record @qualid
+
+   This :term:`table` specifies a set of qualids which are displayed as records.  Use the
+   :cmd:`Add` and :cmd:`Remove` commands to update the set of qualids.
 
-Syntax of Record Projections
+.. table:: Printing Constructor @qualid
 
-  .. insertprodn term_projection term_projection
+   This :term:`table` specifies a set of qualids which are displayed as constructors.  Use the
+   :cmd:`Add` and :cmd:`Remove` commands to update the set of qualids.
 
-  .. prodn::
-     term_projection ::= @term0 .( @qualid {? @univ_annot } {* @arg } )
-     | @term0 .( @ @qualid {? @univ_annot } {* @term1 } )
+.. flag:: Printing Projections
 
-The corresponding grammar rules are given in the preceding grammar. When :token:`qualid`
-denotes a projection, the syntax :n:`@term0.(@qualid)` is equivalent to :n:`@qualid @term0`,
-the syntax :n:`@term0.(@qualid {+ @arg })` to :n:`@qualid {+ @arg } @term0`.
-and the syntax :n:`@term0.(@@qualid {+ @term0 })` to :n:`@@qualid {+ @term0 } @term0`.
-In each case, :token:`term0` is the projected object and the
-other arguments are the parameters of the inductive type.
+   Activates the projection form (dot notation) for printing projections (off by default).
 
-Since the projected object is part of the notation,
-it is always considered an explicit argument of :token:`qualid`,
-even if it is formally declared as implicit (see :ref:`ImplicitArguments`),
+   .. example::
 
+      .. coqtop:: all
 
-.. note:: Records defined with the :cmd:`Record` command are not allowed to be
-   recursive (references to the record's name in the type of its field
-   raises an  error). To define recursive records, one can use the
-   :cmd:`Inductive` and :cmd:`CoInductive` commands, resulting in an inductive or coinductive record.
-   Definition of mutually inductive or coinductive records are also allowed, as long
-   as all of the types in the block are records.
+         Check top half.  (* off: application form *)
+         Set Printing Projections.
+         Check top half.  (* on:  projection form *)
 
-.. note:: Induction schemes are automatically generated for inductive records.
-   Automatic generation of induction schemes for non-recursive records
-   defined with the :cmd:`Record` command can be activated with the
-   :flag:`Nonrecursive Elimination Schemes` flag (see :ref:`proofschemes-induction-principles`).
+.. note:: Records exist in two flavors. In the first
+   implementation, a record :n:`@ident` with parameters :n:`{* @binder }`,
+   constructor :n:`@ident__0`, and fields :n:`{* @name @field_spec }`
+   is represented as a variant type with a single
+   constructor: :n:`Variant @ident {* @binder } : @sort := @ident__0
+   {* ( @name @field_spec ) }` and projections are defined by case analysis.
+   In the second implementation, records have
+   primitive projections: see :ref:`primitive_projections`.
```

Comment:
This note is already present above so should be removed there (maybe a merge confict?):
```suggestion
```